contrib/create-waku/src/cli.ts

```diff
@@ -0,0 +1,153 @@
+        message: 'Project Name',
+        initial: defaultProjectName,
+        onState: (state: any) => (targetDir = String(state.value).trim() || defaultProjectName)
+      },
+      {
+        name: 'shouldOverwrite',
+        type: () => canSafelyOverwrite(targetDir) ? null : 'confirm',
+        message: `${targetDir} is not empty. Remove existing files and continue?`,
+      },
+      {
+        name: 'overwriteChecker',
+        type: (values: any) => {
+          if (values.shouldOverwrite === false) {
+            throw new Error(red('✖') + ' Operation cancelled')
+          }
+          return null
+        }
+      },
+      {
+        name: 'packageName',
+        type: () => (isValidPackageName(targetDir) ? null : 'text'),
+        message: 'Package name',
+        initial: () => toValidPackageName(targetDir),
+        validate: (dir: string) => isValidPackageName(dir) || 'Invalid package.json name'
+      },
+      {
+        name: 'chooseProject',
+        type: 'select',
+        message: 'Choose a starter template',
+        choices: [
+          { title: 'basic-template', value: CHOICES[0] },
+          { title: 'async-template', value: CHOICES[1] },
+          { title: 'promise-template', value: CHOICES[2] },
+        ],
+      }
+    ], {
+      onCancel: () => {
+        throw new Error(red('✖') + ' Operation cancelled')
+      }
+    })
+  } catch (cancelled) {
+    if(cancelled instanceof Error) 
+      console.log(cancelled.message)
+    process.exit(1)
+  }
+
+  const { packageName, shouldOverwrite, chooseProject } = result
+
+  const root = path.join(cwd, targetDir)
+
+  if (shouldOverwrite) {
+    emptyDir(root)
+  } else if (!fs.existsSync(root)) {
+    fs.mkdirSync(root)
+  }
+
+  const pkg = { name: packageName?? toValidPackageName(targetDir), version: '0.0.0' }
+
+  fs.writeFileSync(
+    path.resolve(root, 'package.json'),
+    JSON.stringify(pkg, null, 2),
+  )
+
+  console.log("Setting up project...")
+  const templateRoot = path.resolve(cwd, 'template')
+
+  const render = function render(templateName: string) {
+    const templateDir = path.resolve(templateRoot, templateName)
+    renderTemplate(templateDir, root)
+  }
+
+  render(chooseProject as string)
+
+  const manager = process.env.npm_config_user_agent ?? ''
+  const packageManager = /pnpm/.test(manager)
+    ? 'pnpm'
+    : /yarn/.test(manager)
+      ? 'yarn'
+      : 'npm'
+
+  const commandsMap = {
+    install: {
+      pnpm: 'pnpm install',
+      yarn: 'yarn',
+      npm: 'npm install'
+    },
+    dev: {
+      pnpm: 'pnpm dev',
+      yarn: 'yarn dev',
+      npm: 'npm run dev'
+    }
+  }
+
+  console.log(`\nDone. Now run:\n`)
+  if (root !== cwd) {
+    console.log(`${bold(green(`cd ${path.relative(cwd, root)}`))}`)
+  }
+  console.log(`${bold(green(commandsMap.install[packageManager]))}`)
+  console.log(`${bold(green(commandsMap.dev[packageManager]))}`)
+  console.log()
+}
+
+init().catch((e) => {
+  console.error(e)
+})
```

Comment:
This file feels a little bit complicated for me. Maybe, we look for something in the future.